wish.go

```diff
@@ -59,39 +60,62 @@ func Fatalf(s ssh.Session, f string, v ...interface{}) {
 // STDERR, followed by a new line and an exit 1.
 func Fatalln(s ssh.Session, v ...interface{}) {
 	Errorln(s, v...)
-	Errorf(s, "\r")
 	_ = s.Exit(1)
 	_ = s.Close()
 }
 
 // Error prints the given error the the session's STDERR.
 func Error(s ssh.Session, v ...interface{}) {
-	_, _ = fmt.Fprint(s.Stderr(), v...)
+	str := fmt.Sprint(v...)
+	if _, _, active := s.Pty(); active {
+		str = strings.ReplaceAll(str, "\n", "\r\n")
+	}
+	_, _ = fmt.Fprint(s.Stderr(), str)
 }
 
 // Errorf formats according to the given format and prints to the session's STDERR.
 func Errorf(s ssh.Session, f string, v ...interface{}) {
-	_, _ = fmt.Fprintf(s.Stderr(), f, v...)
+	str := fmt.Sprintf(f, v...)
+	if _, _, active := s.Pty(); active {
+		str = strings.ReplaceAll(str, "\n", "\r\n")
+	}
+	_, _ = fmt.Fprint(s.Stderr(), str)
 }
 
 // Errorf formats according to the default format and prints to the session's STDERR.
 func Errorln(s ssh.Session, v ...interface{}) {
-	_, _ = fmt.Fprintln(s.Stderr(), v...)
+	str := fmt.Sprintln(v...)
+	if _, _, active := s.Pty(); active {
+		str = strings.ReplaceAll(str, "\n", "\r\n")
+	}
+	_, _ = fmt.Fprint(s.Stderr(), str)
 }
 
 // Print writes to the session's STDOUT followed.
 func Print(s ssh.Session, v ...interface{}) {
-	_, _ = fmt.Fprint(s, v...)
+	str := fmt.Sprint(v...)
+	if _, _, active := s.Pty(); active {
+		str = strings.ReplaceAll(str, "\n", "\r\n")
+	}
+	_, _ = fmt.Fprint(s, str)
 }
 
 // Printf formats according to the given format and writes to the session's STDOUT.
 func Printf(s ssh.Session, f string, v ...interface{}) {
-	_, _ = fmt.Fprintf(s, f, v...)
+	str := fmt.Sprintf(f, v...)
+	if _, _, active := s.Pty(); active {
+		str = strings.ReplaceAll(str, "\n", "\r\n")
+	}
+	_, _ = fmt.Fprint(s, str)
 }
 
 // Println formats according to the default format and writes to the session's STDOUT.
 func Println(s ssh.Session, v ...interface{}) {
-	_, _ = fmt.Fprintln(s, v...)
+	str := fmt.Sprintln(v...)
+	if _, _, active := s.Pty(); active {
+		str = strings.ReplaceAll(str, "\n", "\r\n")
+	}
```

Comment:
this could be wrapped in a io.Writer I think, so we don't need to repeat it multiple times...